### PR TITLE
Add custom job for Trivy databases retag

### PR DESCRIPTION
**Pull request by fhielpos**
Towards: https://github.com/giantswarm/giantswarm/issues/31965

This is an experimental PR to add the Trivy databases to our registries and keeping them in sync

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -490,3 +586,13 @@ workflows:
...
     <<: *build_and_retag
+
+  # Runs every 12 hours at 09:08 and 21:08 UTC
+  trivy_databases:
+    triggers:
+      - schedule:
+          cron: "08 09,21 * * *"
+          filters:
+            branches:
+              only: main
```
Comment (fhielpos) [inline .circleci/config.yml:597]: Maybe this is not relevant to have, but better be safe than sorry

File: .circleci/config.yml [hunk 2]
```diff
@@ -129,6 +129,55 @@ jobs:
             - "<<parameters.filepath>>.filtered"
             - "<<parameters.filepath>>.filtered.errlog"
 
+  retag-trivy-dbs:
```
Comment (fhielpos) [inline .circleci/config.yml:132]: This can be re-used for any image, maybe it could have another name

**Comment by fhielpos** [comment]
Since now the job is a cronjob, it won't run on push to this branch. Here is the last workflow run:
https://app.circleci.com/pipelines/github/giantswarm/retagger/3787/workflows/b7b2a8da-4929-4ad5-8461-59f9b0c13fe2

File: .circleci/config.yml (hunk 2 above)
Comment (stone-z) [reply .circleci/config.yml:132]: Why not just call `retag-registry` with a yaml that contains the DB repositories?

Comment (fhielpos) [reply .circleci/config.yml:132]: I tried it, but it requires way too much work to validate the files and it's too prone to fail, so I reverted to this approach which runs faster and doesn't require as many moving pieces.

**Review by stone-z — APPROVED** [review]

**Pull request merged**